software/.clang-tidy

```diff
@@ -37,5 +37,61 @@ Checks: >
 #WarningsAsErrors: '*'
 # This is OK because clang-tidy still displays the warnings
 
-# TODO: Add naming style checks (readability-identifier-naming).
+# Naming style checks (readability-identifier-naming).
+CheckOptions:
+  - key:             readability-identifier-naming.ClassCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.TypeAliasCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.TypedefCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.EnumCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.NamespaceCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.EnumConstantCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.EnumConstantPrefix
```

Comment:
I think the use of  `k` for all constants makes the code harder to read.

Reply:
IMO, any discussion on Hungarian notation should begin with understanding its original intent: https://www.joelonsoftware.com/2005/05/11/making-wrong-code-look-wrong/

Hint: It had nothing to do with types or type qualifiers.

Reply:
> I think the use of  `k` for all constants makes the code harder to read.

I kind of agree with you here. There was an agreement to use googlestyle a while back. This PR is about enforcing it in the code, where it doesn't ruin readability (which IMO is everywhere except the `unit(value)` and `Unit::unit()` functions, which are rendered hard to read by the `CamelCase` convention for function names).
Whether googlestyle is the right tool may still be open to discussion: I don't really have a strong opinion about it, though I do see some benefits, mostly that it makes it easier to know what kind of object we are manipulating.
We could take it further with a convention that tells us what is the nature of the data which that object holds (or, in the case of a function, what nature of operation it does) but that is outside the realm of googlestyle (though inside the broader term "naming convention").

In defense of the `kCamelCase` convention, it makes it obvious (to someone who knows the convention) we are using a constant. We could use something else entirely (`UPPER_CASE` should immediately come to mind to any coder, though it is reserved for a different use in googlestyle).

Also in defense to using googlestyle, I really don't want to create yet another convention for how to write variable/function names, which is completely different (and mostly independent) from "how to name variables/functions?" (for which it seems like I share your views). AFAIK, that second problem can only be enforced by stringent code reviews, not by automated tools.

I believe this discussion belongs to a SW meeting.

Reply:
@dheater excellent article. Glad to have read it.

Looks like this is converging upon a reasonable compromise where actual instances of consts are `k`-prefixed, but at least enum values are not? Because the latter did seem excessive...